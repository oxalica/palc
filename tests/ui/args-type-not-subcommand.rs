#[derive(palc::Args)]
struct Cli1 {
    #[command(subcommand)]
    deep: Deep,
}

struct Deep {}

#[derive(palc::Args)]
struct Cli2 {
    #[command(subcommand)]
    cmd: Subargs,
}

#[derive(palc::Args)]
struct Subargs {}

fn main() {}
