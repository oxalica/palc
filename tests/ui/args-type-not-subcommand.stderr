error[E0277]: `Deep` is not a `palc::Subcommand`
 --> tests/ui/args-type-not-subcommand.rs:4:11
  |
4 |     deep: Deep,
  |           ^^^^ this type is expected to have `derive(palc::Subcommand)` but it is not
  |
  = help: the trait `Subcommand` is not implemented for `Deep`
note: required by a bound in `GetSubcommand::Subcommand`
 --> src/runtime.rs
  |
  |     type Subcommand: Subcommand;
  |                      ^^^^^^^^^^ required by this bound in `GetSubcommand::Subcommand`

error[E0277]: `Subargs` is not a `palc::Subcommand`
  --> tests/ui/args-type-not-subcommand.rs:12:10
   |
12 |     cmd: Subargs,
   |          ^^^^^^^ this type is expected to have `derive(palc::Subcommand)` but it is not
   |
   = help: the trait `Subcommand` is not implemented for `Subargs`
note: required by a bound in `GetSubcommand::Subcommand`
  --> src/runtime.rs
   |
   |     type Subcommand: Subcommand;
   |                      ^^^^^^^^^^ required by this bound in `GetSubcommand::Subcommand`

error[E0277]: `Deep` is not a `palc::Subcommand`
 --> tests/ui/args-type-not-subcommand.rs:4:11
  |
4 |     deep: Deep,
  |           ^^^^ this type is expected to have `derive(palc::Subcommand)` but it is not
  |
  = help: the trait `Subcommand` is not implemented for `Deep`

error[E0277]: `Subargs` is not a `palc::Subcommand`
  --> tests/ui/args-type-not-subcommand.rs:12:10
   |
12 |     cmd: Subargs,
   |          ^^^^^^^ this type is expected to have `derive(palc::Subcommand)` but it is not
   |
   = help: the trait `Subcommand` is not implemented for `Subargs`
